### PR TITLE
[EN] update FaaS & Serverless

### PR DESCRIPTION
**Pull request by johannes-manner**
### Update FaaS and Serverless definitions based on a scientific investigation of these terms

Serverless and FaaS are often used interchangeably when people talk about event-driven stateless functions.
I am a 6th year PhD student at the University of Bamberg, worked a lot on FaaS in the last years as you can see by my [published list of papers](https://www.uni-bamberg.de/pi/team/manner-johannes/).
The ambigous use of terms motivated me to make a structured assessment of scientific literature, how they use the terms FaaS and Serverless. A preprint to my paper can be found [here](https://www.researchgate.net/publication/372214619_A_Structured_Literature_Review_Approach_to_Define_Serverless_Computing_and_Function_as_a_Service).
The reference currently included in the text (Manner 2023) is the landing page (forwarded by `doi.org`) at IEEE as soon as the paper is published - I presented it last week on [IEEE CLOUD 2023](https://conferences.computer.org/cloud/2023/). 

Hope we have a nice discussion here about these two terms :)

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

**Comment by jihoon-seo** [comment]
To reviewers: you can hide whitespace changes with [this link](https://github.com/cncf/glossary/pull/2217/files?w=1). It will make reviewing easier.

File: content/en/function-as-a-service.md [hunk 1]
```diff
@@ -7,30 +7,40 @@ tags: ["infrastructure", "", ""]
...
-In a traditional on-premises scenario, a business manages and maintains its own data center. 
-The business must invest in servers, storage, software, and other technologies 
-and potentially hire an IT staff or contractors to purchase, manage, and upgrade all the equipment and licenses. 
-The data center has to be built to meet peak demand, even when workloads decline and those resources stand idle. 
-Conversely, if the business grows quickly, the IT department might struggle to keep up. 
-Under a standard [Infrastructure-as-a-Service (IaaS)](/infrastructure-as-a-service/) cloud computing model, 
-users pre-purchase capacity units, meaning you pay a public cloud provider for always-on server components to run your apps. 
-It’s the user’s responsibility to scale up server capacity during times of high demand 
-and scale down when that capacity is no longer needed. 
+In a traditional on-premise scenario, a business manages and maintains its own data center.
```
Comment (jihoon-seo) [inline content/en/function-as-a-service.md:26]: In this case `on-premises` is the right use.

```suggestion
In a traditional on-premises scenario, a business manages and maintains its own data center.
```

**Comment by jihoon-seo** [comment]
BTW welcome and thank you for your contribution! Please consider joining [#glossary](https://cloud-native.slack.com/archives/C02TX20MQBB) channel in CNCF Slack too!

**Comment by johannes-manner** [comment]
Can anyone of you, @jihoon-seo, @nate-double-u, @iamNoah1, rerun the Spellcheck CI? :)

**Comment by CathPag** [comment]
Thanks, @johannes-manner. To see where this is in the review pipeline and follow progress, please have a look at the [project board.](https://github.com/orgs/cncf/projects/26/views/1)

File: content/en/serverless.md [hunk 2]
```diff
@@ -7,29 +7,35 @@ tags: ["architecture", "", ""]
...
+All operational aspects like managing physical machines and providing VMs are tasks of the service provider.
+A service provider is typically a public cloud provider but also in-house IT departments can act as service providers for their own development teams.
+They offer interfaces for service users like SDKs, CLIs or OCI compliant runtimes where the focus is on code, configuration options or deploying/uploading artifacts.
+All offerings are pay-per-use.
+[Scaling](/scalability/) and provisioning of additional computing, storage or network resources happens without user interference based on the demand of the deployed applications and ecosystem services like data storage.
+A provider of a serverless platform pools resources and serves multiple users on the same physical machine.
+Tenants are isolated via virtualization approaches in particular [VMs](/virtual-machine/)." [(Manner 2023, p.520)](https://doi.org/10.1109/CLOUD60044.2023.00068)
+
+Serverless is therefore an umbrella term for all services with these characteristics.
+Serverless offerings range from [Platform-as-a-Service (PaaS)](/platform-as-a-service/) until [Software-as-a-Service (SaaS)](/software-as-a-service/).
```
Comment (rolfedh) [inline content/en/serverless.md:20]:
```suggestion
Serverless Computing [abstracts](/abstraction/) servers away from the user.
Operational management falls to the service provider, including handling physical machines and VM provisioning.
Service providers can be public cloud entities or internal IT departments serving their development teams.
These providers offer user interfaces such as SDKs, CLIs, or OCI-compliant runtimes, focusing on code and deployment tasks.
Charges are based on a pay-per-use model.
[Scaling](/scalability/) and resource provisioning for computing, storage, or networking are automatically adjusted based on application demand without user intervention.
A serverless platform provider consolidates resources to serve multiple users on a single physical machine, ensuring isolation through virtualization, especially with [VMs](/virtual-machine/).

Serverless is a comprehensive term encompassing services with these attributes, extending from [Platform-as-a-Service (PaaS)](/platform-as-a-service/) to [Software-as-a-Service (SaaS)](/software-as-a-service/).
```

File: content/en/serverless.md [hunk 3]
```diff
@@ -7,29 +7,35 @@ tags: ["architecture", "", ""]
...
 
-Under a standard [Infrastructure-as-a-Service (IaaS)](/infrastructure-as-a-service/) [cloud computing](/cloud-computing/) model, 
-users pre-purchase units of capacity, meaning you pay a public cloud provider for always-on server components to run your apps. 
-It’s the user’s responsibility to scale up server capacity during times of high demand and 
-to scale down when that capacity is no longer needed. 
+Under a standard [Infrastructure-as-a-Service (IaaS)](/infrastructure-as-a-service/) [cloud computing](/cloud-computing/) model,
+users pre-purchase units of capacity, meaning you pay a public cloud provider for always-on server components to run your services.
+It’s the user’s responsibility to scale up server capacity during times of high demand and
+to scale down when that capacity is no longer needed.
 The cloud infrastructure necessary to run an app is active even when the app isn’t being used.
```
Comment (rolfedh) [inline content/en/serverless.md:28]:
```suggestion
In traditional [Infrastructure-as-a-Service (IaaS)](/infrastructure-as-a-service/) [cloud computing](/cloud-computing/) models, users commit to a predefined capacity, resulting in charges for continuous server availability regardless of actual use.
Responsibility for adjusting server capacity to meet fluctuating demands falls on the user, maintaining active infrastructure even during idle periods.
```

File: content/en/serverless.md [hunk 4]
```diff
@@ -7,29 +7,35 @@ tags: ["architecture", "", ""]
...
+With serverless architecture, by contrast, services are launched only as needed.
+A public cloud provider dynamically allocates resources for these services.
+The user does not have to pay when the service does not run.
+In addition to the cost and efficiency benefits,
+serverless frees developers from routine and menial tasks associated with app scaling and server provisioning.
+With serverless, routine tasks such as managing the operating system and file system, security patches,
+load balancing, capacity management, scaling, logging, and monitoring are all outsourced to a cloud services provider.
+
+See also the glossary for [Function-as-a-Service (FaaS)](/function-as-a-service/).
+Serverless and FaaS are often used as synonyms, but there is a difference.
```
Comment (rolfedh) [inline content/en/serverless.md:41]:
```suggestion
Serverless architecture introduces a more efficient approach, activating services solely upon demand.
This model ensures dynamic resource allocation by a cloud provider, eliminating costs for unused services.
Beyond financial and operational efficiencies, serverless technology relieves developers of the burdens of scaling applications and managing server infrastructure.
Tasks such as operating system maintenance, security updates, load balancing, capacity planning, and monitoring are delegated to the cloud provider, streamlining the development process.

Also see the glossary entry for [Function-as-a-Service (FaaS)](/function-as-a-service/).
Serverless and FaaS are often used as synonyms, but there is a difference.

Refer to the [Function-as-a-Service (FaaS)](/function-as-a-service/) glossary entry for more information.
Although "serverless" and "FaaS" are often used as interchangeable terms, they embody distinct concepts.

```

File: content/en/function-as-a-service.md [hunk 5]
```diff
@@ -7,30 +7,40 @@ tags: ["infrastructure", "", ""]
...
-FaaS infrastructure will auto-scale the code for heavy use, 
-and the developer does not have to spend any time or resources building the code for [scalability](/scalability/). 
+FaaS gives developers an [abstraction](/abstraction/) for running web applications in response to events without managing servers.
+For example, uploading a file could trigger custom code that transcodes the file into various formats.
+FaaS infrastructure will auto-scale the code for heavy use,
+and the developer does not have to spend any time or resources building the code for [scalability](/scalability/).
 Billing is based on computation time alone, which means businesses do not have to pay when the functions are not in use.
+
+See also the glossary for [Serverless](/serverless/).
+Serverless and FaaS are often used as synonyms, but there is a difference.
```
Comment (rolfedh) [inline content/en/function-as-a-service.md:46]:
```suggestion
FaaS provides developers with a platform to execute web applications triggered by events, eliminating the need to manage server infrastructure.
An action, such as file upload, can initiate custom code for tasks like transcoding files into different formats.
The FaaS infrastructure automatically adjusts resources to match demand, freeing developers from the complexities of coding for scalability.
Charges apply solely for the duration of computation, ensuring no costs accrue when functions are inactive.

Refer to the [Serverless](/serverless/) glossary entry for more information.
Although "serverless" and "FaaS" are often used as interchangeable terms, they embody distinct concepts.
```

**Review by rolfedh — COMMENTED** [review]
Great job, Johannes! I've provided a few suggestions to enhance the clarity and brevity of your text. Feel free to use these ideas as you see fit. For better maintainability, consider adding line breaks between sentences instead of within them, as shown in a few instances.

**Comment by nate-double-u** [comment]
Hi @johannes-manner, thanks for your patience. @rolfedh has provided some good feedback - I know it's been a while, but I'd be happy if you're still interested in working through the process of getting your updates in.

**Comment by johannes-manner** [comment]
Hi, I got notified via mail of your pushes :)
Sorry for not responding - was a bussy time the last weeks. 
I would try to look at the changes end of next week and contribute to it. 
@iamNoah1, @nate-double-u is this timeline suitable for you?

**Review by iamNoah1 — APPROVED** [review]
I added the suggestions of @rolfedh and hope to have resolved the merge conflict correctly. Thanks for everyone involved especially @johannes-manner for the first and biggest batch of work.

**Review by nate-double-u — APPROVED** [review]
Thanks for this @johannes-manner and @rolfedh. These are some good changes.

/approve

Based on @johannes-manner's comment above, I'd like to wait a week before actually merging this in case they have some additional comments based on the changes made, but this is very close to landing 🙂

File: content/en/function-as-a-service.md [hunk 6]
```diff
@@ -5,30 +5,25 @@ category: Technology
...
-and potentially hire an IT staff or contractors to purchase, manage, and upgrade all the equipment and licenses. 
-The data center has to be built to meet peak demand, even when workloads decline and those resources stand idle. 
-Conversely, if the business grows quickly, the IT department might struggle to keep up. 
-Under a standard [Infrastructure-as-a-Service (IaaS)](/infrastructure-as-a-service/) cloud computing model, 
-users pre-purchase capacity units, meaning you pay a public cloud provider for always-on server components to run your apps. 
-It’s the user’s responsibility to scale up server capacity during times of high demand 
-and scale down when that capacity is no longer needed. 
-The cloud infrastructure necessary to run an app is active even when the app isn’t being used.
+Traditionally, businesses have relied on maintaining on-premises data centers, necessitating substantial investment in hardware, software, and personnel.
+This setup demands resources to be scaled to peak demand, resulting in underutilized assets during downtimes.
```
Comment (nate-double-u) [inline content/en/function-as-a-service.md:17]:
```suggestion
This setup demands resources to be scaled to peak demand, resulting in underutilized assets during downtime.
```

**Pull request merged**